src/misc/like.ts

```diff
@@ -0,0 +1,243 @@
+ * Set last time of removed like of player
+ */
+function setLikeRemovalTimestamp(player: OfflinePlayer): void {
+  const playerView = dataView(playerLikesType, dataHolder(player));
+  playerView.lastAutoRemoveDate = Date.now();
+}
+
+/*
+ * Get cooldown expiration lastAutoRemoveDate of player
+ */
+function getCooldownTimestamp(player: OfflinePlayer): number {
+  const playerView = dataView(playerLikesType, dataHolder(player));
+  return playerView.cooldownEnds;
+}
+
+/*
+ * Set cooldown expiration lastAutoRemoveDate of player
+ */
+const COOLDOWN_TIME = 24 * 60 * 60 * 1000;
+function setCooldownTimestamp(player: OfflinePlayer): void {
+  const playerView = dataView(playerLikesType, dataHolder(player));
+  playerView.cooldownEnds = Date.now() + COOLDOWN_TIME;
+}
+/*
+ * Add like to player
+ */
+function addLike(sender: CommandSender, uuid: string) {
+  if (!view.playerUuidList) view.playerUuidList = [];
+  const player = uuidToOfflinePlayer(uuid);
+  const playerView = dataView(playerLikesType, dataHolder(player));
+  if (!playerView.count || isNaN(playerView.count)) playerView.count = 0;
+  playerView.count++;
+  sender.sendMessage(`§6Tykkäsit pelaajasta §e${player.name}§6!`);
+  if (view.playerUuidList.includes(uuid)) return;
+  if (view.playerUuidList.length < 10) {
+    view.playerUuidList.push(uuid);
+    return;
+  }
+  const lastUUID = view.playerUuidList[view.playerUuidList.length - 1];
+  const lastPlayer: OfflinePlayer = Bukkit.getOfflinePlayer(
+    UUID.fromString(lastUUID),
+  );
+  if (playerView.count > getLikes(lastPlayer)) {
+    view.playerUuidList[view.playerUuidList.length - 1] = uuid;
+  }
+  view.playerUuidList = sortLikeList(view.playerUuidList);
+}
+
+/*
+ * Remove like from player
+ */
+function removeLike(uuid: string) {
+  if (!view.playerUuidList) view.playerUuidList = [];
+  const player = uuidToOfflinePlayer(uuid);
+  const playerView = dataView(playerLikesType, dataHolder(player));
+  if (!playerView.count || isNaN(playerView.count)) playerView.count = 0;
+  if (playerView.count != 0) {
+    playerView.count--;
+  }
+  if (view.playerUuidList.includes(uuid)) return;
+  if (view.playerUuidList.length < 10) {
+    view.playerUuidList.push(uuid);
+    return;
+  }
+  if (playerView.count == 0) {
+    view.playerUuidList.removeValue(uuid);
+  }
+  view.playerUuidList = sortLikeList(view.playerUuidList);
+}
+
+/**
+ * Has it been enough time to decrease likes
+ * @param timestamp The timestamp, if its 24h ago or more return true
+ */
+const TIME_TO_DECREASE = 24 * 60 * 60 * 1000;
+function hasTimePassedToDecrease(timestamp: number): boolean {
+  return timestamp - (Date.now() - TIME_TO_DECREASE) <= 0;
+}
+
+/**
+ * Calculates milliseconds to hours and minutes
+ * @param duration Milliseconds to be converted
+ */
+function calculateTime(duration: number) {
+  let minutes = Math.floor((duration / (1000 * 60)) % 60),
+    hours = Math.floor((duration / (1000 * 60 * 60)) % 24);
+  hours = hours < 10 ? 0 + hours : hours;
```

Comment:
I did not run the code and might be mistaken, but this *looks* like it is adding 0 to another number instead of string. That obviously wouldn't do anything.